lightkube/core/client.py

```diff
@@ -34,9 +34,11 @@ class Client:
       is ignored when watching changes.
     * **lazy** - When set, the returned objects will be decoded from the JSON payload in a lazy way, i.e. only when
       accessed.
+    * **trust_env** - To ignore environment variables, set trust_env = False
```

Comment:
This name seems to suggest that all env vars will be ignored, but in the case `lightkube.Client()` is initialized without config, it will automatically call `KubeConfig.from_env()` which in turn uses env vars.
I suggest that in the case `trust_env` is set to False and config is not provided, we initialize the config with just `KubeConfig.from_file(DEFAULT_KUBECONFIG)`; this avoids the use of env vars. Will be also good to add a note on this parameter description to point out this behavior.

Reply:
Perhaps i should just improve the description.  I lifted this description directly from [httpx](https://www.python-httpx.org/environment_variables/).  The intent definitely isn't to ignore all environment variables, just the ones pertaining to http_proxy, https_proxy, and no_proxy.

Reply:
The description does help, but still remains the issue that the parameter won't do what is suggested by its name (trust or not the environment) and will look like inconsistent (why trusting the env for some vars but not others?).
My suggesting above will make the parameter behave exactly as expected, and for your specific use case (just ignore proxy env vars) you can initialize the client as follow: 

    client = Client(config=KubeConfig.from_env(), trust_env=False)